web/components/supabase-search.tsx

```diff
@@ -0,0 +1,503 @@
+import { Select } from './widgets/select'
+import { SiteLink } from './widgets/site-link'
+
+const CONTRACTS_PER_PAGE = 20
+
+export const SORTS = [
+  { label: 'Relevance', value: 'relevance' },
+  { label: 'New', value: 'newest' },
+  { label: 'Trending', value: 'score' },
+  { label: 'Daily change', value: 'daily-score' },
+  { label: '24h volume', value: '24-hour-vol' },
+  { label: 'Total traders', value: 'most-popular' },
+  { label: 'High stakes', value: 'liquidity' },
+  { label: 'Last activity', value: 'last-updated' },
+  { label: 'Closing soon', value: 'close-date' },
+  { label: 'Just resolved', value: 'resolve-date' },
+] as const
+
+export type Sort = typeof SORTS[number]['value']
+export const PROB_SORTS = ['prob-descending', 'prob-ascending']
+
+export type filter = 'open' | 'closed' | 'resolved' | 'all'
+
+export type SupabaseSearchParameters = {
+  query: string
+  sort: Sort
+  filter: filter
+  //   facetFilters: SearchOptions['facetFilters']
+}
+
+function getShowTime(sort: Sort) {
+  return sort === 'close-date' || sort === 'resolve-date' ? sort : null
+}
+
+export const INITIAL_STATE = {
+  contracts: undefined,
+  fuzzyContractOffset: 0,
+  shouldLoadMore: true,
+  showTime: null as ShowTime | null,
+}
+
+export type SupabaseAdditionalFilter = {
+  creatorId?: string
+  groupId?: string
+  tag?: string
+  groupSlug?: string
+  excludeContractIds?: string[]
+  excludeGroupSlugs?: string[]
+  excludeUserIds?: string[]
+  nonQueryFacetFilters?: string[]
+}
+
+const AsListContext = createContext({
+  asList: false,
+  setAsList: (_asList: boolean) => {},
+})
+
+export type stateType = {
+  contracts: Contract[] | undefined
+  fuzzyContractOffset: number
+  shouldLoadMore: boolean
+  showTime: ShowTime | null
+}
+
+export function SupabaseContractSearch(props: {
+  defaultSort?: Sort
+  defaultFilter?: filter
+  additionalFilter?: SupabaseAdditionalFilter
+  highlightContractIds?: string[]
+  onContractClick?: (contract: Contract) => void
+  hideOrderSelector?: boolean
+  cardUIOptions?: {
+    hideGroupLink?: boolean
+    hideQuickBet?: boolean
+    noLinkAvatar?: boolean
+  }
+  headerClassName?: string
+  persistPrefix?: string
+  isWholePage?: boolean
+  includeProbSorts?: boolean
+  autoFocus?: boolean
+  profile?: boolean | undefined
+  fromGroupProps?: {
+    group: Group
+    userRole: groupRoleType | null
+  }
+  listViewDisabled?: boolean
+  contractSearchControlsClassName?: string
+}) {
+  const {
+    defaultSort,
+    defaultFilter,
+    additionalFilter,
+    onContractClick,
+    hideOrderSelector,
+    cardUIOptions,
+    highlightContractIds,
+    headerClassName,
+    persistPrefix,
+    includeProbSorts,
+    isWholePage,
+    autoFocus,
+    profile,
+    fromGroupProps,
+    listViewDisabled,
+  } = props
+
+  const [state, setState] = usePersistentState<stateType>(
+    INITIAL_STATE,
+    !persistPrefix
+      ? undefined
+      : { key: `${persistPrefix}-supabase-search`, store: inMemoryStore() }
+  )
```

Comment:
Use `usePersistentInMemoryState` which is the new way.

Hmm, but I guess you're not persisting it without the persistPrefix? I guess you could conditionally use a plain `useState` if we don't want it to persist.

Otherwise, we should make the `persistPrefix` not optional. Optional properties it too easy to mess up if you really should be setting it.